.github/workflows/pipeline.yml

```diff
@@ -78,9 +78,21 @@ jobs:
           name: playwright-a11y-test-results
           path: tests/a11y/playwright-report
 
-      - name: Build an image from Dockerfile
+      - name: Cache Docker image
+        uses: actions/cache@v4
+        with:
+          path: |
+            /tmp/.docker-cache
+          key: ${{ runner.os }}-docker-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-docker-
+
+      - name: Build and cache image from Dockerfile
         run: |
+          mkdir -p /tmp/.docker-cache/${{ env.IMAGE_NAME }}
```

Comment:
```
      - name: Build and cache image from Dockerfile
        if: steps.cache-docker-image.outputs.cache-hit != 'true'
        run: |
          mkdir -p /tmp/.docker-cache/${{ env.IMAGE_NAME }}
```
Couldn't we make this dependent on a cache hit. If we would need to run a test again for any reason, we would not build the image again?

Reply:
Great idea, thank you very much! I didn't think of the rerunning of the pipeline case. I will implement this improvement

Reply:
Your suggestion made me think of another potential improvement: 
If rerunning tests is a common use-case, we could split this into a separate build step so that you can rerun only the tests job.